### PR TITLE
MNT: add workflow to automate wheels build

### LINKED ISSUE #107: ENH: automate wheels build
Let's use cibuildwheel to automate wheels compilation. It should be relatively easy to adapt this workflow from yt's core repo
https://github.com/yt-project/yt/blob/main/.github/workflows/wheels.yaml

### PR DESCRIPTION
**Pull request by neutrinoceros**
Fix #107
I've run the workflow successfully on my fork here https://github.com/neutrinoceros/yt_astro_analysis/actions/runs/1526300758

### CODE REVIEW

File: .github/workflows/wheels.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        with:
+          python-version: '3.9'
+
+      - uses: s-weigand/setup-conda@v1
+        if: matrix.os == 'windows-latest'
+        with:
+          update-conda: true
+          conda-channels: conda-forge
+          activate-conda: true
+          python-version: '3.9'
```
Comment (neutrinoceros) [inline .github/workflows/wheels.yaml:34]: I'm not sure this is relevant in yt-astro, this comes directly from yt, I'll try to simplify it

**Comment by neutrinoceros** [comment]
Note: I have manually validated macOS wheels for
- [ ] CPython 3.7
- [ ] CPython 3.8
- [x] CPython 3.9
- [x] CPython 3.10 (this one takes a couple minutes to install because yt itself doesn't provide wheels for this version yet)

**Comment by neutrinoceros** [comment]
For some reason it seems that I can't install Python 3.7 macOS wheels locally. This is the only supported version for which I'm encountering this issue.
```
ERROR: yt_astro_analysis-1.1.0.dev2-cp37-cp37m-macosx_10_9_x86_64.whl is not a supported wheel on this platform.
```
I also have the exact same problem with latest wheels built for yt itself.

**Comment by brittonsmith** [comment]
> edit: in fact it looks like we _never_ released a `cp37-cp37` wheel for yt itself on PyPI, it was always `cp37-cp37m` (same Python version different ABI), so actually what we're getting here seems to be perfectly in line with yt.

Ok, then I'm inclined to not worry about it. I don't see any reason to go out of our way making wheels for versions without yt wheels. If the user can install yt without the wheel for this Python version, then they should be able to install yt-astro as well.

**Review by brittonsmith — APPROVED** [review]
Excellent, if this is ready to go, please go ahead and merge.

**Pull request merged**